tests/unit/igraph_widest_paths.c

```diff
@@ -0,0 +1,578 @@
+
+    check_invalid_input(&g, &w, &res1, 0, 2, &from, &to, &vertices, &edges,
+                        &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    destroy_vertices_and_edges(&vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    destroy_all(&g, &w, &res1, NULL, &from, &to, NULL, NULL, NULL, NULL, NULL, NULL);
+
+    /* ==================================================================== */
+    /* 4. Empty graph                                                       */
+
+    n = 0;
+    m = 0;
+    igraph_small(&g, n, IGRAPH_UNDIRECTED, -1);
+    igraph_vector_init(&w, m);
+
+    igraph_matrix_init(&res1, n, n);
+    igraph_matrix_init(&res2, n, n);
+    igraph_vs_none(&from);
+    igraph_vs_none(&to);
+
+    run_widest_paths(&g, &w, &res1, &res2, &from, &to, IGRAPH_OUT);
+    /* Should successfully run with nothing occuring */
+
+    destroy_all(&g, &w, &res1, &res2, &from, &to, NULL, NULL, NULL, NULL, NULL, NULL);
+
+    /* ==================================================================== */
+    /* 5. 1 node graph                                                      */
+    printf("=== 5. Testing 1 Node Graph ===\n");
+
+    n = 1;
+    m = 0;
+    igraph_small(&g, n, IGRAPH_UNDIRECTED, -1);
+    igraph_vector_init(&w, m);
+
+    igraph_matrix_init(&res1, n, n);
+    igraph_matrix_init(&res2, n, n);
+    igraph_vs_seq(&from, 0, n-1);
+    igraph_vs_seq(&to, 0, n-1);
+
+    init_vertices_and_edges(n, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    run_widest_paths(&g, &w, &res1, &res2, &from, &to, IGRAPH_OUT);
+    run_get_widest_paths(&g, &w, /* source */ 0, /* destination */ 0, &to, IGRAPH_OUT,
+                            &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+    print_results(n, &res1, &res2, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    destroy_all(&g, &w, &res1, &res2, &from, &to, &vertices, &edges, &predecessors, &inbound_edges,
+                &vertices2, &edges2);
+
+
+    /* ==================================================================== */
+    /* 6. Unreachable Nodes                                                 */
+    printf("\n=== 6. Testing Unreachable Nodes ===\n");
+
+    n = 4;
+    m = 4;
+    igraph_small(&g, n, IGRAPH_DIRECTED, 0, 2, 0, 1, 1, 2, 3, 2, -1);
+    igraph_vector_init_real(&w, m, 1.0, 2.0, 3.0, 5.0);
+
+    igraph_matrix_init(&res1, 1, n);
+    igraph_matrix_init(&res2, 1, n);
+    igraph_vs_1(&from, 0);
+    igraph_vs_seq(&to, 0, n-1);
+
+    init_vertices_and_edges(n, &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    run_widest_paths(&g, &w, &res1, &res2, &from, &to, IGRAPH_OUT);
+    run_get_widest_paths(&g, &w, /* source */ 0, /* destination */ 3, &to, IGRAPH_OUT,
+                            &vertices, &edges, &predecessors, &inbound_edges, &vertices2, &edges2);
+
+    IGRAPH_ASSERT(MATRIX(res1, 0, 3) == IGRAPH_NEGINFINITY);
```

Comment:
There's a few of these assertions which are already checked by comparing the output to the .out file (like the result `[ Inf 2.000000 2.000000 -Inf ]`). I think these can be removed, they don't need a double check.

Reply:
My reasoning was to better highlight the specific purpose of the test for the reviewer, rather than just dump every single output the program makes. 

In hindsight, I agree that the tests are simple enough that the purpose of the test should be obvious. This has been fixed.